compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -657,7 +657,7 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
         else if (tp.classSymbol.is(CaseClass))
```

Comment:
Even if a class is a case class, it might contain a custom unapply, for example:
```scala
abstract class BTypes {
  trait BType

  sealed trait RefBType extends BType {
    def classOrArrayType: String = this match {
      case ClassBType(internalName) => internalName
      case a: ArrayBType            => ""
    }
  }

  final case class ClassBType(val internalName: String) extends RefBType
  class ArrayBType extends RefBType

  object ClassBType {
    def unapply(x: RefBType): Option[String] = None
  }
}
```
Even with this patch, it warns `It would fail on: BTypes.ClassBType(_)`